### DISCUSSION TITLE
AI Research for Maxi Blocks

### DISCUSSION THREAD

**Discussion by ckp267** [maxi-blocks/maxi-blocks#4448] [Ideas]
### Preparations for AI implementations into Maxi Blocks. 

Things to read and learn about. 
AI is moving so fast right now, every week there is a new model or breakthrough. The goal is to just keep reading watching, learning and testing.

I will write an overview soon of how I think it can work with Maxi Blocks. We can use that as a starting point to explore and start thinking about testing small and figure out what creators might like. 

Step one is to see what is going on. Here's a doc of newsletter & twitter sources to follow. Please add to the list if you find new ones.
https://docs.google.com/document/d/1OBFtb7-xPenydFOI2MsS8TEBG1Sw2fAoTPrnjCfS63Q/edit#

Most promising right now for me is #chatyourdata 
Here is one site already live. https://addcontext.xyz/ 

### It might work like this, an AI template:

- Basically, we would make a Maxi Blocks page template for a chatbot front end.
- Api connectors backend. Ingest data - can be from WordPress or elsewhere.
- Modular setup, fill a form to connect data. Embed model.
- Publish live for people to use. Customers that is. 

### How it works; (just make it easy)
- install the WordPress plugin on your website. You get an index of posts, pages, custom post types. You choose what to index from your database. Begin ingestion. 
- set personality, block words, audio watermark, 
- choose a maxi blocks front-end template that works on all devices.
- pick a style card, or give page url we extract your colour scheme
- test your personality
- make it live, offer hosted, takeaway, link with partners
- running costs for api calls are billed monthly, re-ingest at based on required freshness at price per token.
- content income model for buyers, pay per use, send to email, pdf export, email to a friend, export to card templates, so many options.

For Maxi Blocks it becomes a growth driver, every WordPress site is an active installation of Maxi Blocks page builder. 

More to follow...

@Fybex - Just you & me in this discussion right now

**Comment by ckp267** [comment]
Found my old notes from 12 months ago Feb2022. These are just loose ideas. Not well-formed. Putting here to encourage thinking.

It's just an idea, but I think it can be a great service or could help conversions to template library signup or . How can we do is simply? 

### Change template text to creator preference
Imagine you are in the EE Editing Experience. We detect browser language or locality in WP is Spanish. 
Now imaging browsing template or pattern (thumbnails will be English for now) - when you click "preview"  you see this (example) https://maxi-dress.dev200.com/zenith-header/ but the "text placeholders change to fit your language"

**Questions:** 

- How can we do it? 
- Do we need a DB of manual translations linked to the layout? 
- Can we use language model real-time. 

### Change the topic of template text to the user interest. 
"Generate realistic text" > "what is your topic?" "Hairdresser" > Load now. Content changes to be on topic. Use same character length.
- (This might be too much - but they way i see it working is in v2 of Maxi Blocks that we do a micro survey of a topic people are interested in - a bot goes out to find crawl 100 hairdresser website - it comes back with common sitemap - prompt user to pic style card. 
- Then generate the site with urls. Then guide user though page templates - in the style and with filler content in their topic and their language.) THAT would be cool!

Since i had the idea above - ChatGPT came out, and turns our ChatGPT is EXCELLENT for the making a sitemap with titles an all that.
Check this prompt. 

### Let's create a new website flow... What is your topic? "Dentist"
We structure all of this using langchain https://github.com/hwchase17/langchain then we feed the output back to WordPress to create the pages, urls etc.

After page creation - choose a style card- then we load the pages one by one - populate the Maxi Blocks page library and let them choose the templates for each page. 

> This is like a automatic contextually relevant niche site creator that can be exported as a full site editing theme with Maxi Blocks.

**PROMPT:** Create a sitemap outline with 10 pages including; permalinks, seo title (max 80 characters), seo description (max 160 characters), page titles, sub headlines and opening paragraphs for a pizza website.

**OUTPUT:** 

**Home Page**
Permalink: https://www.example.com/
SEO Title: Fresh, Delicious Pizza | Example Pizza
SEO Description: Discover our mouth-watering pizzas made with fresh, high-quality ingredients. Order now and taste the difference at Example Pizza.
Page Title: Welcome to Example Pizza
Sub Headline: Delicious Pizza Made Fresh Daily
Opening Paragraph: Welcome to Example Pizza, where we believe that every pizza should be made with the freshest and highest-quality ingredients. Our pizzas are made fresh daily to ensure that you get the best possible experience. Order now and taste the difference!

**Menu Page**
Permalink: https://www.example.com/menu/
SEO Title: View Our Menu | Example Pizza
SEO Description: Check out our menu of delicious pizzas, sides, and drinks. We offer something for everyone at Example Pizza.
Page Title: Our Menu
Sub Headline: Delicious Pizzas and Sides
Opening Paragraph: At Example Pizza, we believe that everyone deserves a delicious pizza. That's why we offer a wide variety of pizzas, sides, and drinks to satisfy every craving. Check out our menu below!

**Pizza Page**
Permalink: https://www.example.com/pizzas/
SEO Title: Mouth-Watering Pizzas | Example Pizza
SEO Description: Our pizzas are made with fresh, high-quality ingredients and cooked to perfection. Order now and taste the difference at Example Pizza.
Page Title: Our Pizzas
Sub Headline: Fresh and Delicious Pizzas
Opening Paragraph: Our pizzas are made with the freshest and highest-quality ingredients to ensure that you get the best possible experience. Whether you prefer classic toppings or something more adventurous, we have a pizza for you.

**About Us Page**
Permalink: https://www.example.com/about-us/
SEO Title: Our Story | Example Pizza
SEO Description: Learn more about our mission and commitment to serving delicious, high-quality pizzas at Example Pizza.
Page Title: Our Story
Sub Headline: Our Mission and Commitment
Opening Paragraph: At Example Pizza, we are passionate about serving delicious, high-quality pizzas that everyone can enjoy. Learn more about our story, mission, and commitment to quality below.

**Contact Us Page**
Permalink: https://www.example.com/contact-us/
SEO Title: Get in Touch | Example Pizza
SEO Description: Have a question or comment? Get in touch with us today and we'll be happy to help. Contact Example Pizza now.
Page Title: Contact Us
Sub Headline: Reach Out to Us
Opening Paragraph: We love hearing from our customers! Whether you have a question, comment, or suggestion, we're here to help. Get in touch with us below.

**Order Online Page**
Permalink: https://www.example.com/order-online/
SEO Title: Order Online | Example Pizza
SEO Description: Order delicious, high-quality pizzas online from Example Pizza. We offer fast, convenient delivery and pickup options.
Page Title: Order Online
Sub Headline: Fast and Convenient Ordering
Opening Paragraph: Ordering delicious pizzas from Example Pizza has never been easier. Simply click below to get started and enjoy fast, convenient delivery or pickup options.

**Deals and Specials Page**
Permalink: https://www.example.com/deals-and-specials/
SEO Title: Deals and Specials | Example Pizza
SEO Description: Check out our latest deals and specials at Example Pizza. Save money and enjoy delicious pizzas today.
Page Title: Deals and Specials
Sub Headline: Save Money on Delicious

**Comment by Fybex** [comment]
Text that fits the topic is a great idea to start with in the future (looks like the easiest thing for me). Then we also can add something like this for images. The key here is to find the most effective resource to use.  And not just topic for a text and images, all ideas give some directions to think about. I'll start diving deeper into this 👍
Also, @ckp267, grant access to the file please https://docs.google.com/document/d/1OBFtb7-xPenydFOI2MsS8TEBG1Sw2fAoTPrnjCfS63Q/edit#

**Comment by kyrapieterse** [comment]
https://www.supportguy.co/chatbot/UMFDPPIGugxNPhSXj1KR
